SparkPost/2web2ui
# FAD-6466: Reinstate cookie consent timestamp storage

In the related accusers PR, we're giving Heroku and Azure folks the `users/self-manage` grant. The knock-on effect is that the UI will now show those accounts `/accounts/profile`. That is useful for support since ProfilePage shows the account ID. However, the page needs some love to avoid offering capabilities these SSO accounts shouldn't have.

- Make ProfilePage hide some stuff from those accounts using AccessControl
- Ensure we attempt to set the cookie consent flag only once to avoid a loop on error

### Testing
 - Run a local accusers from the hotfix/FAD-6466 branch
 - Fake up a heroku user login (? TBD)
 - Verify that profile page is visible in diminished form
